metadata-ingestion/tests/integration/snowflake/test_snowflake_failures.py

```diff
@@ -330,3 +382,84 @@ def test_snowflake_failed_secure_view_definitions_query_raises_pipeline_warning(
                 ],
             }
         ]
+
+
+# Tests for known_snowflake_edition config option
+@freeze_time(FROZEN_TIME)
+def test_snowflake_is_standard_explicit_true(pytestconfig, snowflake_pipeline_config):
+    snowflake_pipeline_config.source.config.known_snowflake_edition = (
+        SnowflakeEdition.STANDARD
+    )
+
+    with mock.patch("snowflake.connector.connect") as mock_connect:
+        sf_connection = mock.MagicMock()
+        sf_cursor = mock.MagicMock()
+        mock_connect.return_value = sf_connection
+        sf_connection.cursor.return_value = sf_cursor
+
+        sf_cursor.execute.side_effect = query_permission_response_override(
+            default_query_results,
+            [SnowflakeQuery.current_region()],
+            [{"CURRENT_REGION()": "AWS_AP_SOUTH_1"}],
+        )
+
+        pipeline = Pipeline(snowflake_pipeline_config)
+        pipeline.run()
+        pipeline.raise_from_status()
+
+        source = cast(SnowflakeV2Source, pipeline.source)
+        assert source.is_standard_edition() is True
+
+        sf_cursor.execute.assert_any_call(SnowflakeQuery.current_region())
+
+
+@freeze_time(FROZEN_TIME)
+def test_snowflake_is_standard_explicit_false(pytestconfig, snowflake_pipeline_config):
```

Comment:
These tests look largely like they're copy-pasted from each other. In general, that's not a good sign. You might want to look into pytest test parameterization.